packages/lesswrong/lib/types/collectionTypes.ts

```diff
@@ -40,10 +42,10 @@ interface CollectionBase<N extends CollectionNameString = CollectionNameString>
 
   isConnected: () => boolean
 
-  isVoteable: () => this is PgCollection<VoteableCollectionName>
+  isVoteable: () => this is CollectionBase<VoteableCollectionName>;
   makeVoteable: () => void
 
-  hasSlug: () => this is PgCollection<CollectionNameWithSlug>
+  hasSlug: () => boolean
```

Comment:
This caused some weird inference issues due to the whole thing where even though `PgCollection` implements `CollectionBase`, a `PgCollection<N>` isn't necessarily assignable to a `CollectionBase<N>` because of the generic. We can still add the type guard to the collections explicitely in the implementation though.